README.md

```diff
@@ -1,20 +1,89 @@
+4. Choose a method for sizing your icons. In the examples below we're adding CSS class called `rvt-icon`, but you could use `width` and `height` HTML attributes also.
+
+```css
+.rvt-icon {
+  width: 1rem;
+  height: 1rem;
+}
+```
+
+Using **option A** listed above:
+
+```html
+<svg class="rvt-icon">
+  <use xlink:href="rvt-icon-alarm"></use>
+</svg>
+```
+
+Using **option B** listed above:
+
+```html
+<svg class="rvt-icon">
+  <use xlink:href="./path/to/svg/rvt-icons.svg#rvt-icon-alarm"></use>
+</svg>
+```
+
+### Contributing a new icon
+All of the Rivet icon artwork can be found in the Illustrator file in `src/rivet-icons-source.ai`. Each icon is drawn on it's own artboard to the following specifications:
+
 - 16x16px grid
-- 2px stroke
+- 2px stroke for all icon outlines
 - Expand all strokes before exporting and merge/flatten artwork in to one group.
-- set `fill` attribute to `currentColor`
-
-## Docs folder
-Inside the docs folder is a simple static preview page that displays all of the current Rivet icons. Typing `npm run build` into your terminal will generate the demo site and create a preview of every `.svg` icon in the `/svg` folder at build time.
+- set `fill` attribute to `currentColor` on exported SVGs
 
-To add a newly-created icon to the preview page, follow these steps. These instructions assume you have [NodeJS](https://nodejs.org/en/) and NPM installed on your computer.
+#### Adding a new icon
+To add a newly-created icon to the preview page, follow these steps. 
 
+1. Create a new feature branch off of `develop`
 1. Create the new icon artwork in the `/src/rivet-icons-source.ai` file.
-2. Export the icon as an .svg with the strokes outlined and flattened into the `/src/svg` folder.
-3. Type `npm run build` into your terminal to build the preview page.
-4. Commit all changes to the `/src` and `/docs` folder.
+1. Export the icon as an .svg with the strokes outlined and flattened into the `/src/svg` folder.
+1. Follow the instructions above to run the the demo page locally and preview your new icon.
+1. Type `npm run build` into your terminal to build the preview page. This will generate a preview for every file currently in the `/src/svg/` directory.
+1. Commit all changes to the `/src` and `/docs` folder.
+1. Open a pull request against `develop`
+
+#### Demo page
+The Inside the `docs` folder is a simple static preview page that displays all of the current Rivet icons. To run the demo site locally or make updates to the demo site you'll need to do the following. Note: these instructions assume you have [NodeJS](https://nodejs.org/en/) and NPM installed on your computer.
+
+1. Run `npm install` in your terminal to install the necessary dependencies
+1. Type `npm run start` to build a fresh copy of the demo site and start a local development server
+1. Navigate to `http://localhost:8081/` in your web browser to view the demo site.
+
+### Building a custom SVG sprite sheet
```

Comment:
This is an okay solution to start. But it would be better if customizing the sprite file could be done within a build script, rather than a manual process. And what I suggest, if we do it, could easily be done in another PR, rather than extending this one further.

Ideally, a developer could do something like this:

```js
import { buildSprite } from 'rivet-icons'

const spriteString = buildSprite([
  'arrow-up', // Strings first match against the Rivet icon set.
  'arrow-right',
  'arrow-down',
  'arrow-left',
  './src/icons/custom-icon.svg', // Add a single custom icon.
  './src/icons/all-icons-in-this-folder/**.svg' // Add a bunch of icons.
])
```

With `spriteString` (or whatever it is named), the developer could save the output to a file; inline it in HTML; or save it to the app bundle and use JavaScript to inject it in the DOM.

Reply:
Yeah, I like this idea. I'm using a Gulp plugin that wraps the actual `svg-sprite` package just because the API is simpler, but a build custom build script could [use the actual package](https://www.npmjs.com/package/svg-sprite) instead.

It would be neat to create a script like this that get's shipped with the npm package. I'm out of time to work on this for now, but definitely something I'd love to work on. 👍